### DISCUSSION TITLE
Hardware Recommendations

### DISCUSSION THREAD

**Discussion by Lee-Carre** [enzo1982/freac#523] [General Discussion]
In the past (long-time user (from back when Fre:AC was known as BonkEnc; thanks for a well-designed tool, @enzo1982), first time poster) I would simply use whatever optical drive was included with the host.

However, now that especially (new(er)) laptops come *without* an internal optical drive, and dedicated external (USB) drive(s) are required, do users (or contributors/developers with technical insights) have particular (USB) hardware they recommend for quality/accuracy, reliability/endurance, and/or performance?

---

I'm on a small island, so acquiring any specialist tech is non-easy. I did find a local retailer selling [SandBerg-branded rewriter](//www.sandberg.world/product/USB-Mini-DVD-Burner)s (which I needed for other reasons, anyway) for a modest/reasonable price (~£30, if I recall).
Upon interrogation, the drive reports (over USB) itself as
> Slimtype DVD A  DS8A8SH [Rev KC2D]

The USB device `Vendor ID` (`0x152D`) matches [JMicron Technology Corp., who seem to be the manufacturer of the USB–ATAPI bridge](//www.jmicron.com/products) used in the device.

The USB device `Product ID` reported is `0x2329`.

Based on some brief searching on [Amazon](//www.stallman.org/amazon.html), it appears that these drives are somewhat generic, with several different rebrandings; one manufacturer with multiple resellers (which is much more common, in manufacturing, than most people are aware).

Took a gamble, and was pleasantly surprised.
* entirely plug-and-play — no (non-\(generic\|USB-class-compliant\)) drivers, no ‘special’ software
* bootable (eg [POSIX LiveCD](//www.linuxtracker.org/)), without alterations
* slim/compact (I need it to be portable for my use-case); would easily fit in the pocket of a laptop bag
* entirely host-powered — besides power over the regular USB data connection cable (USB-A to USB-B), there's a separate USB-A–to–‘barrel-jack’ for extra power (so, being USB v2 (see below), that's a possible maximum of 1000mA of current draw at 5V); though when only *reading* discs the drive functions without the extra power connected, running by using just the power over the data cable — see below (re M-Disc) for speculation as to why
* multi-format (despite the unclear packaging, it reports (via USB capabilities) to read & (re)write the following):
  * CD-R(W)
  * DVD±R(W)
  * DVD-RAM
  * not mentioned on the packaging, but reports (via USB) of being capable of handling the [M\[illennium\]-Disc](//www.wikipedia.org/wiki/M-DISC) format (which *claims* to be durable for 1000 years) — though, I've not actually tested it thus far. I suspect that higher-power laser output would be needed to write an M-Disc (which seems to use a thicker metallic layer on such discs), and that this is what the extra USB-power cable (mentioned above) is for.
* up to 24× reading/writing (depending on format of the disc; rewriting and more complex (eg DVD-RAM) formats are slower, as expected) — producing FLACs (compression level 8) from a CDDA disc, with all the verification modes enabled (including full CDparanoia), takes ~15–20 minutes (HP EliteBook 840 G3; Intel i5 (4 core), 8GiB RAM); that's with all the performance enhancement modes (such as multi-thread, SuperFast) enabled, too — the thorough verification is the bottleneck, I believe; according to the extraction logs a typical measured/actual extract rate is ~5× (much less than the drive's maximum)
* USB v2.0 bus interface (High-Speed), but that's still plenty of capacity for even a >50× optical drive, let alone a 24× one — I have it connected via a USB v3 hub, without problems
* USB reports `Capabilities` as `0x94` (`Removable`, `UniqueID`, `SurpriseRemovalOK`) — so multiple units of the same drive, connected to the same host, should function without conflict (in honest confession, I haven't yet got around to connecting the second drive I bought)
* `Power State`s implemented/available: `D0`, `D2`, `D3`, `wake from D0`, `wake from D2`
* USB endpoint count is 3
* an [AccurateRip (sample-)offset](http://www.accuraterip.com/driveoffsets.htm) of +6
* under MS Windows v10.0.19045.3324 the drive *does* report when new media is loaded, and correctly responds to software-issued `eject` commands; so semi-automated mass-extraction should be possible
* sturdy (rubber feet), feels reasonably-well built \(for (being of\|having\) polymer (‘plastic’) housing)
* it's proven reliable and problem-free, with moderate use over the several months I've had & used it
* it remains cool, even during sustained operation (eg DVD playback); it becomes slightly warmer than ambient, but **not** hot — the matte-black surface finish probably helps with this — but I have other items on top of it (eg a Flash card reader; I have limited desk surface area) so it's hardly in ideal conditions for cooling, and may well remain at very close to ambient with free airflow
* shaped such that multiple drives are stackable; given the above thermal behaviour, stacking shouldn't pose a problem due to heat, either

I'm willing to supply more (specific) technical data, if anyone wishes.

I was pleased enough that I bought another (for duel-wielding, among other uses), while there were other units of the same model available on the retailer's shelves.

Only downside is that it can be somewhat noisy, when the disc is spun-up, especially to a high RPM. Though, even playback of DVDs (as in, watching cinéma) can be a bit distracting. However, mine happens to be next to me on a non-large desk (and significantly closer than the speakers). Other users may be able to mitigate this by placing it further away, with noise-dampening material between the drive and their ears. Or use [headphones](//www.sennheiser.com/).

Actual power-draw is unclear; with the extra power cable connected, it reports being self-powered, with a (max) draw of 2mA (yes, just two), which doesn't make sense. I have it connected to a decent-spec hub, and it's been very stable, even when extracting rapidly. So, can't be that much in reality.

Being a slimline drive (as used to be in laptops, rather than desktops), it does feature a warning to not press on the top surface of the housing while the drive is in operation.

According to the [AccurateRip drive-offset database](http://www.accuraterip.com/driveoffsets.htm), these types of drives are quite popular.

**Comment by enzo1982** [comment]
Thank you for your post and for choosing fre:ac! It's great that you are still using it after so any years.

The DS-8A8SH is manufactured by Philips Lite-On and I would expect it to be extremely common. If it is reasonably fast, meets your requirements and does not have any obvious flaws (more on that further down), I would say any drive is usable for ripping nowadays.

As AccurateRip compares your ripping results to other users', you can be almost absolutely sure to have a perfect rip if you get a successful verification. At the same time, this allows you to check your drive. Just rip a number of discs that are in the AccurateRip database to see if they verify correctly. If you get mismatches on all tracks of a disc, that's usually just a different pressing or wrong offset. But if only a few tracks fail to verify, that may be a drive issue.

I have two Samsung drives here (SH-D163B and SH-S203N) and the first one often produces bit flips. Ripping the faulty tracks again then usually yields a successful rip, so it's still usable with AccurateRip, but I wouldn't trust it for discs which are not in the AccurateRip database.

Having multiple drives (preferably different models) also allows you to compare rips between the drives when a disc is not in the AccurateRip database. If you get the same results with two different drives, you can be quite sure to have a good rip.

**Reply by Lee-Carre** [reply]
> Thank you […] for choosing fre:ac! It's great that you are still using it after so [m]any years.

It's an easy choice:
* libre-software — both itself, and featuring many libre-codecs (I opt for CDDA music to be output as FLACs, but also work with a lot of human-speech recordings (as files, not from CD, though I also dabble with audiobooks on CDDA); so Opus(/Vorbis) & Speex are very useful)
* focuses on doing one job (as in the [POSIX tradition/philosophy](http://www.catb.org/esr/writings/taoup/))
* has many helpful features — not having only basic (obvious) features, and not for trying to be something beyond audio extraction and/or conversion — this kind of refinement (especially combined with being highly configurable (for differing uses/preferences)) fulfils the ‘doing one job *well*’ part of POSIX philosophy
* easy & pleasant to use (I'm familiar with usability/interaction-design, so see more/deeper than most would, on this); doesn't break (work)flow, feels co-operative instead of awkward or like I'm having to battle with it, and flexible enough to adapt to different uses (one-off extraction of single-disc album, versus semi-automated mass-extraction of many discs)
* (now) has a CLI, for scripting, automating, or using as part of a larger project
* the new(er) feature (since my last meaningful use of Fre:AC) of multiple configuration profiles makes switching use-cases vastly easier, now
 
I was glad to have discovered BonkEnc all those years ago (well over a decade, now). Thankyou for your continued quality work; I notice and appreciate the results.

> The DS-8A8SH is manufactured by Philips Lite-On

Good to know; thanks. I hadn't done much research to learn its origins.

It's encouraging to know that it's from somewhere reputable, with a long history of optical drive design & manufacture (Philips being one of the two original engineering houses behind the CD format, if I recall).

Seems that I was fortunate, this time, in my gamble-like purchase.

> I would expect it to be extremely common.

Yes; string-matching the model code in AccurateRip's DB returns the following lines:

> \# CD Drive	|	Correction Offset	|	Submitted By	|	Percentage Agree
> hp - DVD A DS8A8SH	|	+6	|	223	|	100%
> PLDS - DVD+-RW DS-8A8SH	|	+6	|	128	|	100%
> PLDS - DVD-RW DS8A8SH	|	+6	|	359	|	100%
> Slimtype - DVD A DS8A8SH	|	+6	|	278	|	100%
> Slimtype - DVD A DS8A8SH	|	+6	|	45	|	100%

There are other drives with similar model-codes, with even higher usage/report counts.

I notice that some of the most popular drive models have over 1000 users submitting data to AccurateRip. So, while not *the most* popular, these Philips Lite-On drives likely feature in the top-20 list of popularity.

---

> […] I would say any drive is usable for ripping nowadays.
> 
> As AccurateRip compares your ripping results to other users', you can be almost absolutely sure to have a perfect rip if you get a successful verification.

Ah, that brings clarity to the purpose of AccurateRip (even having read a little about it); hardware abstraction/commoditisation. Thankyou for connecting those dots. It's been a while, before 2023, since I did any CDDA extraction, so I've been out of the loop on such developments.

Ironically, that makes this thread somewhat moot. But, perhaps now users can, instead, focus on other criteria of products, if accuracy is now a mostly-solved problem.

While I'm aware that the documentation is very much a work-in-progress, might I suggest that for new users a ‘getting started’ guide — 
else, UI warnings/messages (which appear only for newer users; so either appearing only N times, or with a checkbox to not show again, or possibly considering setting the default to true for messages about a disc failing AccurateRip verification (due to the drive being uncalibrated)) — should mention the importance of, early on, finding a disc which is in the AccurateRip DB (perhaps with a pointer/link to the relevant page listing albums in the AR DB), in order to confirm their drive's offset, **before** they begin mass-extracting their entire CD collection. This would avoid having to re-extract if the user only later discovers the utility/purpose of AccurateRip.

> I have two Samsung drives here (SH-D163B and SH-S203N) and the first one often produces bit flips. Ripping the faulty tracks again then usually yields a successful rip, so it's still usable with AccurateRip, but I wouldn't trust it for discs which are not in the AccurateRip database.

Interesting. Thanks. Bit-flipping sounds like a strange problem for a drive to exhibit. I have mixed opinions about [Samsung](//www.youtube.com/@rossmanngroup/search?query=samsung) in general; notably their anti-repair attitude (as per [Louis Rossmann](//www.youtube.com/@rossmanngroup) of Right to Repair fame).

I, personally, would prefer a drive which may be slower, but which ultimately produces an accurate output on the first attempt without needing repeat extractions. Set-and-forget, almost. Hence why I have full CDparanoia mode enabled by default; if the disc can be reliably read anyway, then CDparanoia doesn't add much work/time, it only really kicks in for imperfect discs.
I'm reminded, years ago, of having borrowed a CD which turned out to be quite badly scratched (the owner neglected to keep them in proper storage; either jewel cases or a CD-wallet), which I had (a now old version of) Fre:AC work on; the extraction (using full CDparanoia; no AccurateRip or other verification back then) took several hours (7 or 8 comes to mind, but at least 4 hours) due to how extensively damaged the disc was. The resulting files weren't terrible, either; listenable, with only occasional silences. I was suitably impressed. I probably caused a lot of wear on the drive, that day.
My thinking being; do it correctly, once, preserve the FLACs (HDD storage has come a long way, in my time), put the CDs into archival storage, and never have to re-extract again (FLACs can always be converted to a newer version of the format).

This is exactly why I was prompted to start this thread; because odd little details like this never make it into the marketing materials for products. Often not even the datasheets (if one knows to search for them). Hence my surprise, when discovering that my take-a-gamble-and-hope-for-the-best purchase of what was available in a local retailer, sans thorough research, turned out to **not** have some hidden snag/catch, and essentially be problem-free. Seemed worth sharing, especially as optical drives (seem destined to) become less common.

> Having multiple drives (preferably different models) also allows you to compare rips between the drives when a disc is not in the AccurateRip database. If you get the same results with two different drives, you can be quite sure to have a good rip.

A good point. I hadn't considered that. Though, I did buy the drive(s) before installing Fre:AC on this host.

I have \(encountered\|access to\) several discs which seem to be in no databases (CDDB, MusicBrainz, let alone AccurateRip). So, hope to contribute the metadata, in future.

**Comment by Lee-Carre** [comment]
Related:
* [Hydrogen Audio § CD Hardware/Software](//www.hydrogenaud.io/index.php/board,20.0.html)
* From #394: [Audio CD ripping — optical drive accuracy listing](//www.pilabor.com/blog/2022/10/audio-cd-ripping-hardware/)

**Comment by Lee-Carre** [comment]
[Fantastic DIY Speakers for less than $30!](//www.youtube.com/watch?v=zdkyGDqU7xA&list=PLzrI14lOlSqeCgDH9zJb-qRKem9fTT2Tj)
